lib/view_component/base.rb

```diff
@@ -28,6 +28,8 @@ class Base < ActionView::Base
     class_attribute :content_areas
     self.content_areas = [] # class_attribute:default doesn't work until Rails 5.2
 
+    attr_accessor :original_view_context
```

Comment:
Is this threadsafe? I wonder if CurrentAttributes would be a more Rails-y way to pull this off.

Reply:
Thinking about it a bit more, I don't think it's much different than the existing code since we set `@view_context` as a property on the component in `render_in`, both aren't thread safe.